### ISSUE TITLE
GH Action for Intel compiler test

### ISSUE THREAD

**Issue by stenczelt** [libAtoms/QUIP#328]
Given that Intel's compilers and maths libraries are available from apt-get now (Intel OneAPI) we can easily test the compilations of the code with them on GitHub Actions as well. Shall we write an action for this, so that we are not only testing GNU?

**Comment by jameskermode** [comment]
yes, good idea - the only reason we are not testing on Intel is that I thought their compilers weren't available on GHA.

**Comment by stenczelt** [comment]
we can simply use their docker images, the HPC-kit should have all we need: `intel/oneapi-hpckit`
